services/net/transcription/TranscriptionManager.cs

```diff
@@ -0,0 +1,206 @@
+                    {
+                        this.Logger.LogInformation("Consuming topic: {topic}", topic);
+
+                        // TODO: Need to learn how to safely stop listening without losing content.  Every time a service goes down it will most likely lose content.
+                        await this.Consumer.ListenAsync<string, SourceContent>(HandleMessageAsync, topic);
+
+                        // Successful run clears any errors.
+                        this.State.ResetFailures();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    this.Logger.LogError(ex, "Service had an unexpected failure.");
+                    this.State.RecordFailure();
+                }
+            }
+
+            // The delay ensures we don't have a run away thread.
+            this.Logger.LogDebug("Service sleeping for {delay} ms", delay);
+            await Task.Delay(delay);
+        }
+    }
+
+    /// <summary>
+    /// Retrieve a file from storage and send to Microsoft Cognitive Services. Obtain
+    /// the transcription and update the content record accordingly.
+    /// </summary>
+    /// <param name="result"></param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    private async Task HandleMessageAsync(ConsumeResult<string, SourceContent> result)
+    {
+        if (this.State.Status != ServiceStatus.Running)
+        {
+            this.Consumer.Stop();
+            this.State.Stop();
+        }
+
+        this.Logger.LogInformation("Updating transcription from: {Topic}, Uid: {Key}", result.Topic, result.Message.Key);
+
+        var content = await _api.FindContentByUidAsync(result.Value.Uid, result.Value.Source);
+
+        // Upload the file to the API.
+        if (content != null && !String.IsNullOrWhiteSpace(result.Value.FilePath))
+        {
+            // TODO: Handle different storage locations.
+            // Remote storage locations may not be easily accessible by this service.
+            // TODO: This allows the captured stream to be converted to text - fix.
+            var sourcePath = Path.Join(_options.ClipPath, result.Value.FilePath);
+            if (File.Exists(sourcePath))
+            {
+                var fileBytes = File.ReadAllBytes(sourcePath);
+                var transcript = await GetTranscriptionAsync(fileBytes);
+                content.Transcription = transcript;
+
+                content = await _api.UpdateContentAsync(content);
+                this.Logger.LogDebug("Content Updated.  Content ID: {Id}", content?.Id);
+            }
+            else
+            {
+                this.Logger.LogError("File not found.  Content ID: {Id}, File: {sourcePath}", content.Id, sourcePath);
+            }
+        }
+        else
+        {
+            this.Logger.LogDebug("Content already exists. Content Source: {Source}, UID: {Uid}", content.Source, content.Uid);
```

Comment:
This message doesn't make sense based on the if statement logic.

Reply:
Fixed.